install/kubernetes/cilium/templates/hubble-relay/deployment.yaml

```diff
@@ -71,11 +71,11 @@ spec:
               protocol: TCP
             {{- end }}
           readinessProbe:
-            tcpSocket:
-              port: grpc
+            exec:
```

Comment:
> Afaik, Cilium moved away from `exec` specifically because it is much more expensive than a network-based probe

Yep, we moved away from `exec` based liveness/readiness probes mainly because of the resource overhead.

Reply:
The issue is that the inbuilt gRPC check doesn't support TLS. But I could use the gRPC check if TLS is disabled and if we're running a recent enough Kubernetes version, and fall back to exec otherwise?